game-core/src/main/java/games/strategy/net/PlayerNameAssigner.java

```diff
@@ -22,37 +21,42 @@
    * for the second node with the name "foo", this method will return "foo (1)".
    *
    * @param desiredName The name being requested by a new user joining.
-   * @param ipAddress The IP address of the user that is joining. Used to determine if the user is
+   * @param mac The (hashed) mac of the user that is joining. Used to determine if the user is
    *     already connected under a different name.
-   * @param loggedInNodes Collection of nodes that have already joined.
    */
   public static String assignName(
       final String desiredName,
-      final InetAddress ipAddress,
-      final Collection<INode> loggedInNodes) {
-    Preconditions.checkNotNull(ipAddress);
-    Preconditions.checkNotNull(loggedInNodes);
+      final String mac,
+      final Multimap<String, String> loggedInMacsToNames) {
     Preconditions.checkArgument(PlayerNameValidation.serverSideValidate(desiredName) == null);
+    Preconditions.checkNotNull(loggedInMacsToNames);
 
-    String currentName = findLoggedInName(ipAddress, loggedInNodes).orElse(desiredName);
-    if (currentName.length() > 50) {
-      currentName = currentName.substring(0, 50);
+    String currentName = desiredName;
+
+    if (!isBotName(desiredName)) {
+      if (currentName.length() > 50) {
```

Comment:
The name truncation now happens before `findExistingLoggedInNameFromSameMac`; Technically a bug fix, by doing the truncation after `findLoggedInName`, the name de-dupe in `findLoggedInName` would not have matched the truncated name (which previously occurred after `findLoggedInName`)